src/engine/Util/Detector.ts

```diff
@@ -1,13 +1,134 @@
 /// <reference path="Log.ts" />
+
+/**
+ * This is the list of features that will be used to log the supported
+ * features to the console when Detector.logBrowserFeatures() is called.
+ */
+const REPORTED_FEATURES: Object = {
+   webgl: 'WebGL',
+   webaudio: 'WebAudio',
+   gamepadapi: 'Gamepad API'
+};
+
 module ex {
 
    /**
+    * Interface for detected browser features matrix
+    */
+   export interface IDetectedFeatures {
+      canvas: boolean;
+      arraybuffer: boolean;
+      dataurl: boolean;
+      objecturl: boolean;
+      rgba: boolean;
+      webaudio: boolean;
+      webgl: boolean;
+      gamepadapi: boolean;
+   }
+
+   /**
     * Excalibur internal feature detection helper class
     */
    export class Detector {
-      
+
+      private _features: IDetectedFeatures = null;
+
       public failedTests: string[] = [];
+
+      public constructor() {
+         this._features = this._loadBrowserFeatures();
+      }
+
+      /**
+       * Returns a map of currently supported browser features. This method
+       * treats the features as a singleton and will only calculate feature
+       * support if it has not previously been done.
+       */
+      public getBrowserFeatures(): Object {
```

Comment:
Updated the type, good catch. I'm torn on using `get` here, even if only for clarity. Referencing `this.getBrowserFeatures` as a property seems dirty to me, somehow. Since this is an exposed method that we're documenting, **AND** it doesn't actually get invoked unless you're in DEBUG mode, I think I'd prefer the optimization of implicit caching. 

If you're not in DEBUG mode, `this._loadBrowserFeatures()` doesn't execute until you call `Detector.logBrowserFeature()` or `Detector.getBrowserFeatures()` explicitly.